### PR TITLE
updated adapter and added pi scrubbing logic

### PR DESCRIPTION
**Pull request by mithunjohny**
# PR Details

### Description

We are in need to update the chat adapter dependency, for that we need to update how we scrub the pi data before sending it to telemetry

## Solution Proposed

Added code to scrub the pi data before sending it to telemetry, created new function which will parse the given input with the known pi keys.

### Acceptance criteria

_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests

- [ ] You have tested all changes in Popout mode
- [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [x] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

**Review by elopezanaya — APPROVED** [review]

File: __tests__/utils/loggers.spec.ts [hunk 1]
```diff
@@ -308,6 +308,56 @@ describe('loggers', () => {
...
+            telemetry.info.mockClear();
+            const eventData = {
+                Event: '',
+                CustomProperties: {
+                    "threadId": "19:acsV1_QmJ5hZE2YVI0b-7oPltD-2WXaI_tQv30JdnoU1KyR3w1@thread.v2",
+                    "sender": {
+                        "kind": "communicationUser",
+                        "communicationUserId": "8:acs:453332fc-94e7-403f-82a0-36f184b6412e_00000023-8893-99f9-df68-563a0d008263"
+                    },
+                    "senderDisplayName": "Mithun Chackankulam",
```
Comment (xTEddie) [inline __tests__/utils/loggers.spec.ts:323]: I'd use a generic name

File: src/utils/loggerUtils.ts [hunk 2]
```diff
@@ -46,10 +58,102 @@ export const setChatId = (chatId: string, ocSdkLogger?: OCSDKLogger | null, acsC
...
+        } else {
+            const disallowedValue = containsDisallowedValues(input);
+            if (disallowedValue) {
+                if (input.length > 1) {
+                    input = `${disallowedValue}:${input.slice(0, 1)}**${input.length - 1} hidden**`;
+                }
+            }
+        }
+    }
+    if (input && typeof input == "object") {
```
Comment (xTEddie) [inline src/utils/loggerUtils.ts:90]: `input === "object"` instead of `input == "object"`?

File: src/utils/loggerUtils.ts [hunk 3]
```diff
@@ -46,10 +58,102 @@ export const setChatId = (chatId: string, ocSdkLogger?: OCSDKLogger | null, acsC
...
+                    if (typeof value == "string") {
+                        // Even if it is a string, it could be a JSON string, so we need to check that first
+                        if (isJsonObject(value)) {
+                            cloned[key] = redactPII(JSON.parse(value), true, false, knownPIIKeys);
+                        } else {
+                            if (value.length > 1) {
+                                cloned[key] = value.slice(0, 1) + `**${value.length - 1} hidden**`;
+                            }
+                        }
+                    } else if (typeof value == "object") {
```
Comment (xTEddie) [inline src/utils/loggerUtils.ts:118]: `input === "object"` instead of `input == "object"`?

File: src/utils/loggerUtils.ts [hunk 4]
```diff
@@ -46,10 +58,102 @@ export const setChatId = (chatId: string, ocSdkLogger?: OCSDKLogger | null, acsC
...
+                        // Found a PII key, but it contains an object, redact all nested keys instead
+                        redactPII(value, true, false, knownPIIKeys);
+                    }
+                } else if (typeof value == "object") {
```
Comment (xTEddie) [inline src/utils/loggerUtils.ts:122]: `input === "object"` instead of `input == "object"`?

File: src/utils/loggerUtils.ts [hunk 5]
```diff
@@ -46,10 +58,102 @@ export const setChatId = (chatId: string, ocSdkLogger?: OCSDKLogger | null, acsC
...
+            }
+
+            for (const [key,value] of Object.entries(cloned)) {
+                // Convert all keys to lower case to simplify matching
+                const keyLowerCase = key.toLowerCase();
+
+                if (redactAllNested || knownPIIKeys.includes(keyLowerCase)) {
+                    // Found a PII key, redact it. Not currently considering UUID, or number types as PII
+                    // Any key that contains "token" is considered PII
+                    if (typeof value == "string") {
```
Comment (xTEddie) [inline src/utils/loggerUtils.ts:109]: `input === "string"` instead of `input == "string"`?

File: src/utils/loggerUtils.ts [hunk 6]
```diff
@@ -46,10 +58,102 @@ export const setChatId = (chatId: string, ocSdkLogger?: OCSDKLogger | null, acsC
...
+        }
+    }
+    return undefined;
+}
+
+export function redactPII(input: unknown, redactAllNested = false, shouldClone = true, knownPIIKeys = _basePIIKeys): unknown {
+    if (input === undefined || input === null) {
+        return input;
+    }
+    if (typeof input == "boolean") {
```
Comment (xTEddie) [inline src/utils/loggerUtils.ts:74]: `input === "boolean"` instead of `input == "boolean"`?

File: src/utils/loggerUtils.ts (hunk 5 above)
Comment (mithunjohny) [reply src/utils/loggerUtils.ts:109]: done

File: src/utils/loggerUtils.ts (hunk 6 above)
Comment (mithunjohny) [reply src/utils/loggerUtils.ts:74]: done

File: src/utils/loggerUtils.ts (hunk 4 above)
Comment (mithunjohny) [reply src/utils/loggerUtils.ts:122]: done

File: src/utils/loggerUtils.ts (hunk 2 above)
Comment (mithunjohny) [reply src/utils/loggerUtils.ts:90]: done

**Review by xTEddie — APPROVED** [review]

**Review by elopezanaya — APPROVED** [review]
